internal/rbd/mirror.go

```diff
@@ -33,6 +33,39 @@ func (ri *rbdImage) EnableImageMirroring(mode librbd.ImageMirrorMode) error {
 	}
 	defer image.Close()
 
+	parentInfo, err := image.GetParent()
+	if err != nil {
+		return fmt.Errorf("failed to get parent info of image %q: %w",
+			ri, err)
+	}
+
+	if parentInfo.Image.Trash {
+		return fmt.Errorf("failed to enable mirroring on image %q:"+
+			" parent %q is in trash",
+			parentInfo.Image.ImageID, ri)
+	}
+
+	parent, err := ri.getParent()
```

Comment:
lets add `Trash` to the `rbdImage` struct and make only one call `getParent()` (need to populate Trash value as well) which gives all the details. we don't need to call `getParent` multiple times.